src/main/scala/com/codacy/clangtidy/ParserConfig.scala

```diff
@@ -0,0 +1,36 @@
+package com.codacy.clangtidy
+
+import scopt.OptionParser
+
+import scala.io.Codec
+
+case class ParserConfig(encoding: Codec = Codec.default)
+
+object ParserConfig {
+
+  def parse(toolName: String): OptionParser[ParserConfig] = new OptionParser[ParserConfig](toolName) {
+    opt[String]('e', "encoding")
+      .action { (encoding, config) =>
+        val newCodec = try {
+          Codec(encoding)
+        } catch {
+          case e: java.nio.charset.UnsupportedCharsetException =>
+            System.err.println(s"Unsupported encoding: $encoding")
+            throw e
+        }
+        config.copy(encoding = newCodec)
+      }
+      .text(s"input encoding (default: ${ParserConfig().encoding.name})")
+  }
+
+  def withConfig(toolName: String, args: Array[String])(f: ParserConfig => Unit) = {
```

Comment:
I think it's better to return `Int` here and avoid `System.exit(1)` inside a helper function.
Return the return value and then in the `main` you can `System.exit` with that value.